src/app/core/common-components/entities-table/entities-table.component.ts

```diff
@@ -0,0 +1,354 @@
+  @Output() filteredRecordsChange = new EventEmitter<T[]>(true);
+
+  private updateFilteredData() {
+    this.addActiveInactiveFilter(this._filter);
+    const filterPredicate = this.filterService.getFilterPredicate(this._filter);
+    const filteredData = this._records.filter(filterPredicate);
+    this.recordsDataSource.data = filteredData.map((record) => ({ record }));
+
+    this.filteredRecordsChange.emit(filteredData);
+  }
+
+  @Input() set filterFreetext(value: string) {
+    this.recordsDataSource.filter = value;
+  }
+
+  /** function returns the background color for each row*/
+  @Input() getBackgroundColor?: (rec: T) => string = (rec: T) => rec.getColor();
+  idForSavingPagination: string;
+
+  @Input() clickMode: "popup" | "navigate" | "none" = "popup";
+  @Output() rowClick: EventEmitter<T> = new EventEmitter<T>();
+
+  /**
+   *
+   *
+   * BULK SELECT
+   * User can use checkboxes to select multiple rows, so that parent components can execute bulk actions on them.
+   *
+   *
+   */
+  @Input() set selectable(v: boolean) {
+    this._selectable = v;
+    this.columnsToDisplay = this._columnsToDisplay;
+  }
+  _selectable: boolean = false;
+
+  readonly ACTIONCOLUMN_SELECT = "__select";
+
+  /**
+   * outputs an event containing an array of currently selected records (checkmarked by the user)
+   * Checkboxes to select rows are only displayed if you set "selectable" also.
+   */
+  @Output() selectedRecordsChange: EventEmitter<T[]> = new EventEmitter<T[]>();
+  @Input() selectedRecords: T[] = [];
+
+  selectRow(row: TableRow<T>, event: MatCheckboxChange) {
+    if (event.checked) {
+      this.selectedRecords.push(row.record);
+    } else {
+      const index = this.selectedRecords.indexOf(row.record);
+      if (index > -1) {
+        this.selectedRecords.splice(index, 1);
+      }
+    }
+
+    this.selectedRecordsChange.emit(this.selectedRecords);
+  }
+
+  /**
+   * INLINE EDIT
+   * User can switch a row into edit mode to change and save field values directly from within the table
+   */
+  @Input() set editable(v: boolean) {
+    this._editable = v;
+    this.columnsToDisplay = this._columnsToDisplay;
+  }
+  _editable: boolean = true;
+  readonly ACTIONCOLUMN_EDIT = "__edit";
+  /**
+   * factory method to create a new instance of the displayed Entity type
+   * used when the user adds a new entity to the list.
+   */
+  @Input() newRecordFactory: () => T;
+
+  /**
+   * Show one record's details in a modal dialog (if configured).
+   * @param row The entity whose details should be displayed.
+   */
+  onRowClick(row: TableRow<T>) {
+    if (!row.formGroup || row.formGroup.disabled) {
+      this.showEntity(row.record);
+    }
+    this.rowClick.emit(row.record);
+  }
+
+  showEntity(entity: T) {
+    switch (this.clickMode) {
+      case "popup":
+        this.formDialog.openFormPopup(entity, this.columnsToDisplay); // TODO this.formDialog.openFormPopup(entity, this._columns)
```

Comment:
we do not correctly support the following flags yet:
- hideFromTable?: boolean;
- hideFromForm?: boolean;
- forTable?: boolean;

Reply:
--> discuss: where + how do we need these concepts; let's try to simplify and clearly specify this before re-adding the logic.

(e.g. todos-related-to-entity use the process)

Reply:
we discussed to move towards a different config structure in the future:
- config for list columns
- config for details fields (popup or fullscreen) separately
- config for entity, including custom (run-time calculated) field configs

![image](https://github.com/Aam-Digital/ndb-core/assets/1682541/01d09fc0-63e1-4826-9c68-4cff91d3a970)


Might only refactor things towards this at a later point, however.